### PR TITLE
Fixes #8291 - make foreman_url pluggable

### PR DESCRIPTION
**Pull request by dustints**
depends on https://github.com/theforeman/foreman/pull/1904

### CODE REVIEW

**Comment by dustints** [comment]
struggling to figure out how to fix this test.

getting...

```
undefined method `split' for false:FalseClass (NoMethodError)
/var/lib/workspace/workspace/test_develop_pr_core/database/mysql/ruby/1.9.3/test/unit/orchestration/tftp_test.rb:33
```

it's getting false for `generate_pxe_template` on a host...not sure how it's getting a bool back.
The test passes on my machine. anybody have any ideas to fix this test?

**Comment by dustints** [comment]
one of the tests was setting a constant that was affecting other tests in the suite.

**Comment by dLobatog** [comment]
@dustints This needs a rebase

**Comment by dustints** [comment]
@eLobato, thanks --- rebased.

**Comment by unorthodoxgeek** [comment]
@dustints please rebase

File: app/controllers/concerns/foreman/controller/foreman_url_renderable.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    def override?(host)
+      @provider.override?(host)
+    end
+
+    def foreman_url(*args)
+      if @provider.present?
+        @provider.foreman_url(*args)
+      else
+        raise 'provider missing'
```
Comment (lzap) [inline app/controllers/concerns/foreman/controller/foreman_url_renderable.rb:17]: This must be ForemanException, properly formatted. Also expand on which provider are we erroring out about.

http://projects.theforeman.org/projects/foreman/wiki/Translating#Exceptions

File: app/controllers/concerns/foreman/controller/renderer.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        end
+      end
+
+      def snippet(name, options = {})
+        if (template = ConfigTemplate.where(:name => name, :snippet => true).first)
+          logger.debug "rendering snippet #{template.name}"
+          begin
+            return unattended_render(template)
+          rescue => exc
+            raise "The snippet '#{name}' threw an error: #{exc}"
```
Comment (lzap) [inline app/controllers/concerns/foreman/controller/renderer.rb:50]: Ditto, there are others as well.

**Comment by lzap** [comment]
Out of my curiosity, why you need to override this?

**Comment by lzap** [comment]
Sorry for the delay, once you rebase for the last time and fix the comments above, I am going to test this immediately. Why you need this feature again please? Is this for Katello plugin?

**Pull request closed without merging**